File: graduates/suhyun/2week/BOJ_1924.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.time.LocalDate;
import java.util.StringTokenizer;

public class BOJ_1924 {
    // 범용성 따지면 해당 코드가 좋음 시간은 두 코드 시간은 동일하고 메모리차이도 비슷
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int month = Integer.parseInt(st.nextToken());
        int day = Integer.parseInt(st.nextToken());

        int[] daysInMonth = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
        String[] weekdays = {"SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT"};

        int sum = 0;

        for (int i = 0; i < month - 1; i++) {
            sum += daysInMonth[i];
        }
        sum += day;
        System.out.println(weekdays[sum % 7]);

    }

    public static void main2(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int x = Integer.parseInt(st.nextToken());
        int y = Integer.parseInt(st.nextToken());

        LocalDate localDate = LocalDate.of(2007, x, y);
        String result = localDate.getDayOfWeek().toString();
        System.out.println(result.substring(0, 3));
    }
}

File: sehun/2024/2week/BOJ_1924.py
x, y = map(int, input().split())

days_of_week = ["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"]

# 각 month의 월요일 영점조절용(?)
# ex)   1월1일에서 다음월로 이동하면 기존 커서? 좌표?의 '일'에 아래의 값을 더해준다.
#       1월에서 2월로 이동시 --> 1일 + days_of_month[1]값을 일에다 더해주기 -> 2월 1+4일이 첫월요일

days_of_month = {
    1: 4,
    2: 0,
    3: 4,
    4: 5,
    5: 4,
    6: 5,
    7: 4,
    8: 4,
    9: 5,
    10: 4,
    11: 5,
    12: 4,
}
month = 1
day = 1

# 각 월의 첫번째주 월요일의 날짜 구하기
for i in range(0, x - 1):
    month += 1
    day += days_of_month[i + 1]
    if day > 7:
        day = day % 7

index = (y - day) % 7

print(days_of_week[index])
